synapse/storage/databases/main/e2e_room_keys.py

```diff
@@ -328,26 +374,26 @@ def _get_e2e_room_keys_version_info_txn(txn):
         )
 
     @trace
-    async def create_e2e_room_keys_version(self, user_id: str, info: dict) -> str:
+    async def create_e2e_room_keys_version(self, user_id: str, info: JsonDict) -> str:
         """Atomically creates a new version of this user's e2e_room_keys store
         with the given version info.
 
         Args:
-            user_id(str): the user whose backup we're creating a version
-            info(dict): the info about the backup version to be created
+            user_id: the user whose backup we're creating a version
+            info: the info about the backup version to be created
 
         Returns:
             The newly created version ID
         """
 
-        def _create_e2e_room_keys_version_txn(txn):
+        def _create_e2e_room_keys_version_txn(txn: LoggingTransaction) -> str:
             txn.execute(
                 "SELECT MAX(version) FROM e2e_room_keys_versions WHERE user_id=?",
                 (user_id,),
             )
-            current_version = txn.fetchone()[0]
+            current_version = cast(Tuple[Optional[int]], txn.fetchone())[0]
             if current_version is None:
-                current_version = "0"
+                current_version = 0
 
             new_version = str(int(current_version) + 1)
```

Comment:
Do we still need the call to `int` here?

Edit: oh, there's some weird strings/int stuff going on here. I think I expanded on this in another comment.

Reply:
I noticed that this line is strictly incorrect: `new_version` ought to be an `int` here because we insert it into a bigint column in the database.

Reply:
Good spot, the cast to `int` is redundant.